Web Inspector: Update CSSDocumentation.js with upstream data. January 2023

rcaliman-apple opened the pull request:
#### cd77ac52ebd261c83be029e8d83746259f67f07d
<pre>
Web Inspector: Update CSSDocumentation.js with upstream data. January 2023
<a href="https://bugs.webkit.org/show_bug.cgi?id=251150">https://bugs.webkit.org/show_bug.cgi?id=251150</a>

Reviewed by Patrick Angle.

Sync `CSSDocumentation.js` with latest data from upstream source:
<a href="https://github.com/microsoft/vscode-custom-data">https://github.com/microsoft/vscode-custom-data</a>

Notable changes:
- ampersands in descriptions are no longer encoded in the upstream data.
- some existing properties got MDN reference urls.
- new properties added: `animation-composition`, `caret`, `caret-shape`,
`contain-intrinsic-*` properties, `scroll-timeline-*` properites, `math-depth`, `math-shift`.

None of the overrides have been addressed upstream since the last sync. They remain.

New `font-synthesis-*` properties were added to overrides in PR #4329 but they don&apos;t exist upstream.
Updated the script to account for this scenario and include them in the output.

* Source/WebInspectorUI/Scripts/update-inspector-css-documentation:
* Source/WebInspectorUI/UserInterface/External/CSSDocumentation/CSSDocumentation-overrides.json:

Fixed format of reference URLs in overrides to match upstream format.

* Source/WebInspectorUI/UserInterface/External/CSSDocumentation/CSSDocumentation.js:

Canonical link: <a href="https://commits.webkit.org/259369@main">https://commits.webkit.org/259369@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/74fd49b81f4ea8e9cc59621efbe1c4e9243f6a8e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/104723 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/13800 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/37626 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/113998 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/174200 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/108639 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/14929 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/4732 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/97054 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/112924 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/110483 "Passed tests") | [⏳ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/iOS-16-Simulator-WK2-Tests-EWS "Waiting to run tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/94561 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/39075 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/108192 "Passed tests") | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/26175 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/80742 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/1/builds/94653 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/7154 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/27531 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/92615 "Built successfully") | 
| [  ~~🛠 🧪 jsc-arm64~~](https://ews-build.webkit.org/#/builders/83/builds/4903 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/7259 "Built successfully") | [⏳ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Ventura-Debug-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/30174 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/103544 "Passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/13308 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/47091 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/101302 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6457 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/9047 "Built successfully") | | [  ~~🧪 jsc-mips-tests~~](https://ews-build.webkit.org/#/builders/45/builds/25243 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

patrickangle commented on Source/WebInspectorUI/UserInterface/External/CSSDocumentation/CSSDocumentation.js:
```diff
@@ -503,42 +511,42 @@ CSSDocumentation = {
         "url": "https://developer.mozilla.org/docs/Web/CSS/border-block-color"
     },
     "border-block-end": {
-        "description": "Logical 'border-bottom'. Mapping depends on the parent element\u2019s 'writing-mode', 'direction', and 'text-orientation'.",
+        "description": "Logical 'border-bottom'. Mapping depends on the parent element's 'writing-mode', 'direction', and 'text-orientation'.",
```
Just to confirm, this is the result of the upstream data changing, right? It's unfortunate we are loosing smart quotes for non code things...

rcaliman-apple replied:
Yes, that's coming from upstream, not from our handling in the script.

rcaliman-apple replied:
Occurred in this change upstream: https://github.com/microsoft/vscode-custom-data/commit/549576c0e32f9944536e4505a82fa1077a666f2d

rcaliman-apple committed:
Web Inspector: Update CSSDocumentation.js with upstream data. January 2023
https://bugs.webkit.org/show_bug.cgi?id=251150

Reviewed by Patrick Angle.

Sync `CSSDocumentation.js` with latest data from upstream source:
https://github.com/microsoft/vscode-custom-data

Notable changes:
- ampersands in descriptions are no longer encoded in the upstream data.
- some existing properties got MDN reference urls.
- new properties added: `animation-composition`, `caret`, `caret-shape`,
`contain-intrinsic-*` properties, `scroll-timeline-*` properites, `math-depth`, `math-shift`.

None of the overrides have been addressed upstream since the last sync. They remain.

New `font-synthesis-*` properties were added to overrides in PR #4329 but they don't exist upstream.
Updated the script to account for this scenario and include them in the output.

* Source/WebInspectorUI/Scripts/update-inspector-css-documentation:
* Source/WebInspectorUI/UserInterface/External/CSSDocumentation/CSSDocumentation-overrides.json:

Fixed format of reference URLs in overrides to match upstream format.

* Source/WebInspectorUI/UserInterface/External/CSSDocumentation/CSSDocumentation.js:

Canonical link: https://commits.webkit.org/259369@main
